engine/execution/provider/engine.go

```diff
@@ -315,12 +315,20 @@ func (e *Engine) deliverChunkDataResponse(chunkDataPack *flow.ChunkDataPack, req
 	// sends requested chunk data pack to the requester
 	deliveryStartTime := time.Now()
 
+	nonce, err := rand.Uint64()
```

Comment:
Please add:
- error handling as necessary
- TODO to address log+swallow pattern

Reply:
https://github.com/onflow/flow-go/pull/4258/commits/7c1d3d9c89b0dff40e17882828d19f0d85d62333